axum/src/extract/query.rs

```diff
@@ -137,4 +151,35 @@ mod tests {
         let res = client.get("/?n=hi").send().await;
         assert_eq!(res.status(), StatusCode::BAD_REQUEST);
     }
+
+    #[test]
+    fn test_try_from_uri() {
+        #[derive(Deserialize)]
+        struct TestQueryParams {
+            foo: String,
+            bar: u32,
+        }
+        let uri: Uri = "http://example.com/path?foo=hello&bar=42".parse().unwrap();
+        let query: Result<Query<TestQueryParams>, _> = Query::try_from(&uri);
+        assert!(query.is_ok());
+
+        let query = query.unwrap();
```

Comment:
```suggestion
        let query: Query<TestQueryParams> = Query::try_from(&uri).unwrap();
```

Nit: an `.unwrap()` is essentially equivalent to `assert!(query.is_ok())`

Reply:
Gotcha, that makes sense